### PR TITLE
[pdcurses] Support UNIX, use vcpkg_build_nmake on Windows

### LINKED ISSUE #14620: [<port name>] build failure
**Host Environment**
- OS: macOS
- Compiler: revision

**To Reproduce**
`./vcpkg install pdcurses`

**Failure logs** 
./vcpkg install pdcurses
Computing installation plan...
The following packages will be built and installed:
    pdcurses[core]:x64-osx
Detecting compiler hash for triplet x64-osx...
Starting package 1/1: pdcurses:x64-osx
Building package pdcurses[core]:x64-osx...
Could not locate cached archive: /Users/shawntschoepe/.cache/vcpkg/archives/2c/2ca683e8098d51e342d89fa93b975aaa49a896ad.zip
-- Using cached /Users/shawntschoepe/dev/vcpkg/downloads/wmcbrine-PDCurses-6c1f95c4fa9f9f105879c2d99dd72a5bf335c046.tar.gz
-- Cleaning sources at /Users/shawntschoepe/dev/vcpkg/buildtrees/pdcurses/src/5bf335c046-43b71d7fd1.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/shawntschoepe/dev/vcpkg/downloads/wmcbrine-PDCurses-6c1f95c4fa9f9f105879c2d99dd72a5bf335c046.tar.gz
-- Using source at /Users/shawntschoepe/dev/vcpkg/buildtrees/pdcurses/src/5bf335c046-43b71d7fd1.clean
-- Build x64-osx-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:106 (message):
    Command failed: NMAKE-NOTFOUND /A -f /Users/shawntschoepe/dev/vcpkg/buildtrees/pdcurses/src/5bf335c046-43b71d7fd1.clean/wincon/Makefile.vc WIDE=Y UTF8=Y
    Working Directory: /Users/shawntschoepe/dev/vcpkg/buildtrees/pdcurses/src/5bf335c046-43b71d7fd1.clean/wincon
    Error code: No such file or directory
    See logs for more information:

Call Stack (most recent call first):
  ports/pdcurses/portfile.cmake:23 (vcpkg_execute_required_process)
  scripts/ports.cmake:136 (include)

Error: Building package pdcurses:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: pdcurses:x64-osx
  Vcpkg version: 2020.11.12-unknownhash

Additionally, attach any relevant sections from the log files above.

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

### PR DESCRIPTION
**Pull request by NancyLi1013**
**Describe the pull request**

- What does your PR fix? Fixes #14620

Add `Supports` field 

Add unsupported message

Update ci.baseline.txt

Currently, vcpkg only adds `wincon `module to compile, so it can only support Windows platform.
https://github.com/wmcbrine/PDCurses/blob/master/wincon/README.md

For uwp, it will post the error like this:

`
LINK : fatal error LNK1104: cannot open file 'LIBCMT.lib'`

Note: No feature needs to test.

### CODE REVIEW

File: ports/pdcurses/portfile.cmake [hunk 1]
```diff
@@ -1,3 +1,5 @@
+vcpkg_fail_port_ibnstall(ON_TARGET "uwp" "linux" "osx")
```
Comment (NancyLi1013) [inline ports/pdcurses/portfile.cmake:1]:
```suggestion
vcpkg_fail_port_install(ON_TARGET "uwp" "linux" "osx")
```

File: ports/pdcurses/portfile.cmake [hunk 2]
```diff
@@ -70,4 +72,4 @@ endif()
 
 file(INSTALL ${CMAKE_CURRENT_LIST_DIR}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/pdcurses RENAME copyright)
```
Comment (NancyLi1013) [inline ports/pdcurses/portfile.cmake:73]:
```suggestion
file(INSTALL ${CMAKE_CURRENT_LIST_DIR}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```

**Comment by JackBoosY** [comment]
Close this PR temporary.

**Pull request closed without merging**